cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/table/field/FieldType.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spi.data.table.field;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Contains the name and type information of a {@link Field}.
+ */
+public final class FieldType {
+  /**
+   * Supported data types.
+   */
+  public enum Type {
+    INTEGER,
+    LONG,
+    FLOAT,
+    DOUBLE,
+    STRING
+  }
+
+  static final Set<Type> PRIMARY_KEY_TYPES = ImmutableSet.of(Type.INTEGER, Type.LONG, Type.STRING);
```

Comment:
Wonder if having primary key information with `Type` enum would be better. Something like:

```java
  public enum Type {
    INTEGER(true),
    LONG(true),
    FLOAT(false),
    DOUBLE(false),
    STRING(true)
    
   private final boolean isPrimaryKeyType;

    Type(boolean isPrimaryKeyType) {
      this.isPrimaryKeyType = isPrimaryKeyType;
    }

    public boolean isPrimaryKeyType() {
      return isPrimaryKeyType;
    }
  }
```

That way there is no need to keep this set and have additional method `isPrimaryKeyType` in `Fields` class.

Reply:
Adding parameters to Type can easily get complicated. Let's say we restrict the types for indexes tomorrow, then we'll have to add another parameter. Keeping a separate list will allow us to have add more lists easily. What do you think?